src/Backups/BackupImpl.cpp

```diff
@@ -929,6 +935,12 @@ void BackupImpl::writeFile(const BackupFileInfo & info, BackupEntryPtr entry)
         writer->copyDataToFile(info.data_file_name, create_read_buffer, info.base_size, info.size - info.base_size);
     }
 
+    if (!deduplicate_files)
+    {
+        for (const auto & reference : info.reference_sources)
+            writer->copyFile(reference, info.data_file_name, info.size - info.base_size);
```

Comment:
This code won't work for archives. It's not difficult to fix it for archives though.

Reply:
hmm I'm not 100% sure about reading and writing to same archive

Reply:
We can copy the same file multiple times. I know that's not optimal, but archive backups are slow anyway (and we don't use them for S3 because writing an archive is always single-threaded).

Reply:
> from the source to different destinations

If by source you mean generating from entries then yes it is easy.

Reply:
I meant executing [those lines](https://github.com/ClickHouse/ClickHouse/pull/56460/files/9d965368a2a337ef6cc2566462670e26fc9e2799#diff-4848814683263bfb8784e5382bc1aff6e7075617c74451174b49f1819905374eR919-R924) for each element from `info.data_file_copies` too.

Reply:
okay we are on the same page, that makes sense to me.